### PR TITLE
Unbreak paperspace

### PR DESCRIPTION
**Pull request by freezeboy**
###### Motivation for this change

Add several python packages to be able to unbreak paperspace

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by omasanori** [comment]
Result of `nixpkgs-review pr 102424` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python27Packages.gradient_statsd</li>
  </ul>
</details>
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>python27Packages.click-help-colors</li>
    <li>python37Packages.click-help-colors</li>
    <li>python37Packages.gradient_sdk</li>
    <li>python37Packages.gradient_statsd</li>
    <li>python37Packages.hyperopt</li>
    <li>python37Packages.paperspace</li>
    <li>python38Packages.click-help-colors</li>
    <li>python38Packages.gradient_sdk</li>
    <li>python38Packages.gradient_statsd</li>
    <li>python38Packages.hyperopt</li>
    <li>python38Packages.paperspace</li>
  </ul>
</details>

File: pkgs/development/python-modules/gradient_statsd/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "iWlNX43ZtvU73wz4+8DgDulQNOnssJGxTBkvAaLj530=";
+  };
+
+  propagatedBuildInputs = [ requests datadog ];
+
+  # tries to use /homeless-shelter to mimic container usage, etc
+  #doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/gradient_statsd/default.nix:17]: Is this a mistake? If you disable the tests can you add an importCheck?

File: pkgs/development/python-modules/hyperopt/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  version = "0.1.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "34xIpivBYUvcN/XMVwBkqTpLgaFlWWIduazuP2U2tlg=";
+  };
+
+  propagatedBuildInputs = [ future cloudpickle numpy networkx six tqdm scipy pymongo ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/hyperopt/default.nix:17]: Why are the tests disabled? We should probably replace it with importCheck.

File: pkgs/development/python-modules/click-help-colors/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "EZ5fr2nPyRnJlcWWIyasj9h/EeVqNxr1lOPf2EWPTG4=";
+  };
+
+  propagatedBuildInputs = [ click ];
+
+  # tries to use /homeless-shelter to mimic container usage, etc
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/click-help-colors/default.nix:17]: Can you do importCheck instead?

**Comment by freezeboy** [comment]
Fixed the broken package and added suggested tests

**Comment by omasanori** [comment]
Result of `nixpkgs-review pr 102424` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python27Packages.click-help-colors</li>
    <li>python27Packages.gradient_statsd</li>
    <li>python37Packages.click-help-colors</li>
    <li>python37Packages.gradient_sdk</li>
    <li>python37Packages.gradient_statsd</li>
    <li>python37Packages.hyperopt</li>
    <li>python37Packages.paperspace</li>
    <li>python38Packages.click-help-colors</li>
    <li>python38Packages.gradient_sdk</li>
    <li>python38Packages.gradient_statsd</li>
    <li>python38Packages.hyperopt</li>
    <li>python38Packages.paperspace</li>
  </ul>
</details>

**Review by omasanori — APPROVED** [review]

File: pkgs/development/python-modules/click-help-colors/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchPypi, buildPythonPackage
```
Comment (omasanori) [inline pkgs/development/python-modules/click-help-colors/default.nix:1]:
```suggestion
{ stdenv, fetchPypi, buildPythonPackage, isPy3k
```

File: pkgs/development/python-modules/click-help-colors/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+, click, pytest
+}:
+
+buildPythonPackage rec {
+  pname = "click-help-colors";
+  version = "0.8";
+
```
Comment (omasanori) [inline pkgs/development/python-modules/click-help-colors/default.nix:8]:
```suggestion

   disabled = !isPy3k;

```

File: pkgs/development/python-modules/gradient_statsd/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{ lib, stdenv, fetchPypi, buildPythonPackage
```
Comment (omasanori) [inline pkgs/development/python-modules/gradient_statsd/default.nix:1]:
```suggestion
{ lib, stdenv, fetchPypi, buildPythonPackage, isPy3k
```

File: pkgs/development/python-modules/gradient_statsd/default.nix [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+, boto3, requests, datadog, configparser, python
+}:
+
+buildPythonPackage rec {
+  pname = "gradient_statsd";
+  version = "1.0.1";
+
```
Comment (omasanori) [inline pkgs/development/python-modules/gradient_statsd/default.nix:8]:
```suggestion

    disabled = !isPy3k;

```

File: pkgs/development/python-modules/gradient_statsd/default.nix [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "iWlNX43ZtvU73wz4+8DgDulQNOnssJGxTBkvAaLj530=";
+  };
+
+  propagatedBuildInputs = [ requests datadog ]
+    ++ lib.optional python.isPy2 configparser;
```
Comment (omasanori) [inline pkgs/development/python-modules/gradient_statsd/default.nix:15]:
```suggestion
  propagatedBuildInputs = [ requests datadog ];
```

**Review by omasanori — COMMENTED** [review]
While the current changes work, I'd suggest to disable Python 2 builds instead of fixing them.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 102424 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>12 packages built:</summary>  <ul>    <li>python27Packages.click-help-colors</li>    <li>python27Packages.gradient_statsd</li>    <li>python37Packages.click-help-colors</li>    <li>python37Packages.gradient_sdk</li>    <li>python37Packages.gradient_statsd</li>    <li>python37Packages.hyperopt</li>    <li>python37Packages.paperspace</li>    <li>python38Packages.click-help-colors</li>    <li>python38Packages.gradient_sdk</li>    <li>python38Packages.gradient_statsd</li>    <li>python38Packages.hyperopt</li>    <li>python38Packages.paperspace</li>  </ul></details>

**Comment by jonringer** [comment]
@omasanori he already did the work to make those builds work. But you're right, python2 is probably not a worthwhile platform to support.

**Comment by omasanori** [comment]
@jonringer Sure, it makes sense to merge anyway.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102424` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>python27Packages.gradient_statsd</li>
    <li>python37Packages.gradient_statsd</li>
    <li>python37Packages.paperspace</li>
    <li>python38Packages.gradient_statsd</li>
    <li>python38Packages.paperspace</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>python27Packages.click-help-colors</li>
    <li>python37Packages.click-help-colors</li>
    <li>python37Packages.gradient_sdk</li>
    <li>python37Packages.hyperopt</li>
    <li>python38Packages.click-help-colors</li>
    <li>python38Packages.gradient_sdk</li>
    <li>python38Packages.hyperopt</li>
  </ul>
</details>

**Pull request merged**